README.md

```diff
@@ -74,39 +91,96 @@ An optional parameter `key` has been added that a particular install of prebid c
+Prebid Cache will use the custom keys for those elements that include them and will create system-generated `uuid`s for the ones that don't. Note that if configuration flag `allow_setting_keys` is set to `false` or simply not set inside the `config.yaml` file, Prebid Cache would generate random `uuid`s for all elements and not use the custom keys `"CustomKeyValueHere"` nor `"AnotherCustomKeyValue"` at all.
 
-### GET /cache?uuid={id}
+#### Overwritting values
+
+Prebid Cache does not allow overwritting any value, for either autogenerated or custom keys. If an entry already exists for a given key, it will not be overwitten, and an empty string will be returned as the `uuid` value of that entry. Suppose we wanted to overwrite the entries under `"CustomKeyValueHere"` and the system-generated key `"147c9934-894b-4c1f-9a32-e7bb9cd15376"`.
+
+```json
+{
+  "puts": [
+    {
+      "type": "xml",
+      "ttlseconds": 60,
+      "value": "<tag>Some other XML content under a key already in use.</tag>",
+      "key": "CustomKeyValueHere"
+    },
+    {
+      "type": "json",
+      "ttlseconds": 300,
+      "value": [2, false, "Trying to store other JSON under a key already in use."],
+      "key": "147c9934-894b-4c1f-9a32-e7bb9cd15376"
+    },
+    {
+      "type": "xml",
+      "ttlseconds": 30,
+      "value": "<tag>Different XML content will be stored under a system-generated key.</tag>",
+    }
+  ]
+}
+```
 
-Retrieves a single value from the cache. If the `id` isn't recognized, then it will return a 404.
+Then we get a response with empty `uuid` fields for those `"puts"` elements that tried to overwrite entries:
 
-Assuming the above POST calls have been made, here are some sample GET responses.
+```json
+{
+  "responses": [
+    {"uuid": ""},
+    {"uuid": ""},
+    {"uuid": "efc6ca1d-3409-4b8b-96e5-aec508a57639"}
+  ]
+}
+```
+
+Those items in the cache did not get overwritten as a subsequent `GET` call would assert:
+
+```
+$ curl http://someprebidcachehost.com/cache\?uuid\=CustomKeyValueHere
+
+HTTP/1.1 200 OK
+Content-Type: application/xml
+
+<tag>Your XML content goes here.</tag>
+```
 
----
+This is to prevent bad actors from trying to overwrite legitimate caches with malicious content, or a poorly coded app overwriting its own cache with new values, generating uncertainty of what is actually stored under a particular key. Note that cases like these are the only where a subset of caches would not get stored. Under any other scenario, we expect the entire request to fail and no elements to get stored.
```

Comment:
Note that cases like these are the only **_time_** where a subset of caches would not get stored.